tools/common/FileCopier.cs

```diff
@@ -123,6 +123,13 @@ static CopyFileResult CopyFileCallback (CopyFileWhat what, CopyFileStep stage, I
 					else if (stage == CopyFileStep.Err) {
 						Log (1, "Could not copy the file '{0}' to '{1}'", source, target);
 						return CopyFileResult.Quit;
+					} else if (stage == CopyFileStep.Start) {
+						if (File.Exists (target) || Directory.Exists (target)) {
+							Log (1, "Deleted target {0}, it's not up-to-date", target);
+							// This callback won't be called for directories, but we can get here for symlinks to directories.
+							// This means that File.Delete should always work (no need to check for a directory to call Directory.Delete)
+							File.Delete (target);
+						}
```

Comment:
I don't thin that it the target is a dir the delete will work. I did a quick tests in the csharp RELP and I get the following exception:


```csharp
var dir = "Mydir"; 
Directory.CreateDirectory(dir); 
File.Delete(dir);
```     

Exception is thrown:
                                                                                                                                               

> System.UnauthorizedAccessException: Access to the path '/Users/mandel/Desktop/Mydir' is denied. ---> System.IO.IOException: Operation not permitted                                     --- End of inner exception stack trace ---                                                                                                                                          at System.IO.FileSystem.DeleteFile (System.String fullPath) [0x0006d] in <dd436d8b1aa7421eb3bcf48b995163c4>:0                                                                        at System.IO.File.Delete (System.String path) [0x00014] in <dd436d8b1aa7421eb3bcf48b995163c4>:0                                                                                      at <InteractiveExpressionClass>.Host (System.Object& $retval) [0x00000] in <7680ce88366e47b0aaff572764d5938d>:0                                                                      at Mono.CSharp.Evaluator.Evaluate (System.String input, System.Object& result, System.Boolean& result_set) [0x00038] in <6f83349cf15e4f8ab74a5f73daa81e78>:0                         at Mono.CSharpShell.Evaluate (System.String input) [0x00000] in <1a1d94369c5b4ba08b4121dc5d616b7f>:0

Reply:
My understanding, from the comment, is that it’s not called for directories.
But it’s called for a symlink - that can point to a directory and the symlink is actually a file (to delete).

Reply:
Ok, in that case, I don't think we need the Exists at all. The File.Delete already handles if the file exists or not. Directory.Exist is just needed to confirm that is a symlink, but since File.Delete will do the right thing, I believe we can remove the if and simply do a File.Delete(target)